### PR TITLE
Fix math/fixed floor/ceil/round

### PR DESCRIPTION
**Pull request by edyu**
This fixes https://github.com/odin-lang/Odin/issues/3080 especially when negative number is involved.

### CODE REVIEW

File: core/math/fixed/fixed.odin [hunk 1]
```diff
@@ -1,55 +1,61 @@
...
-Fixed :: struct($Backing: typeid, $Fraction_Width: uint)
-	where
-		intrinsics.type_is_integer(Backing),
-		0 <= Fraction_Width,
-		Fraction_Width <= 8*size_of(Backing) {
-	i: Backing,
+Fixed :: struct($Backing: typeid,
+$Fraction_Width: uint,) where intrinsics.type_is_integer(Backing),
+	0 <=
+	Fraction_Width,
```
Comment (Kelimion) [inline core/math/fixed/fixed.odin:11]: This is unnecessarily hard to read. Why was this reformatted?

File: core/math/fixed/fixed.odin [hunk 2]
```diff
@@ -1,55 +1,61 @@
...
+	x.i &= 1 << Fraction_Width - 1
 	x.i |= integer
 }
 
 to_f64 :: proc(x: $T/Fixed($Backing, $Fraction_Width)) -> f64 {
 	sign := -1.0 if x.i < 0 else 1.0
 	num := math.abs(x.i)
 	res := f64(num >> Fraction_Width)
-	res += f64(num & (1<<Fraction_Width-1)) / f64(1<<Fraction_Width)
+	res += f64(num & (1 << Fraction_Width - 1)) / f64(1 << Fraction_Width)
```
Comment (Kelimion) [inline core/math/fixed/fixed.odin:58]: In fact there's a whole lot of reformatting going on that's nothing to do with fixing a bug.

File: core/math/fixed/fixed.odin [hunk 3]
```diff
@@ -88,21 +94,19 @@ div_sat :: proc(x, y: $T/Fixed($Backing, $Fraction_Width)) -> (z: T) {
 
 @(require_results)
 floor :: proc(x: $T/Fixed($Backing, $Fraction_Width)) -> Backing {
-	return x.i >> Fraction_Width
+	if x.i >= 0 do return x.i >> Fraction_Width
+	else do return (x.i - (1 << (Fraction_Width - 1)) + (1 << (Fraction_Width - 2))) >> Fraction_Width
```
Comment (Kelimion) [inline core/math/fixed/fixed.odin:98]: We don't use `do` in the core packages.

File: core/math/fixed/fixed.odin [hunk 4]
```diff
@@ -88,17 +88,20 @@ div_sat :: proc(x, y: $T/Fixed($Backing, $Fraction_Width)) -> (z: T) {
 
 @(require_results)
 floor :: proc(x: $T/Fixed($Backing, $Fraction_Width)) -> Backing {
-	return x.i >> Fraction_Width
+	if x.i >= 0 {
+		return x.i >> Fraction_Width
+	}
```
Comment (flysand7) [inline core/math/fixed/fixed.odin:93]: The brace alignmtnt in use by the core library is to put else on the same line as the closing brace of the `if`, by the way

**Comment by edyu** [comment]
https://github.com/odin-lang/Odin/pull/3082/commits/7f6f97128492092b82af6298e1c3d863dea07518

**Comment by edyu** [comment]
@Kelimion @flysand7 when you have chance, please review/approve this merge. Thank you.

**Pull request merged**